python/GafferSceneTest/PrimitiveVariableQueryTest.py

```diff
@@ -0,0 +1,656 @@
+			self.names,
+			self.plugTypes,
+			self.defaultValues
+		) :
+			qnv.addQuery( plugType( defaultValue=defaultValue ), name + str(interpolation) + self.nonVectorSuffix )
+
+		# check queries (non-vector data)
+
+		for i, ( dataType, value ) in enumerate( zip( self.dataTypes, self.values ) ) :
+			output = qnv["out"][ i ]
+			self.assertTrue( output["exists"].getValue() )
+			self.assertEqual( output["interpolation"].getValue(), interpolation )
+			self.assertEqual( output["type"].getValue(), dataType.staticTypeName() )
+			self.assertEqual( output["value"].getValue(), value )
+
+		# add nodes to script node
+
+		scriptNode = Gaffer.ScriptNode()
+		scriptNode.addChild( quadScene )
+		scriptNode.addChild( qnv )
+		scriptNode.addChild( qv )
+
+		# check queries
+
+		for i, interpolation in enumerate( self.interpolations ) :
+			for j, ( vectorDataType, value ) in enumerate( zip( self.vectorDataTypes, self.values ) ) :
+				output = scriptNode["qv"]["out"][ i * len( self.names ) + j ]
+				self.assertTrue( output["exists"].getValue() )
+				self.assertEqual( output["interpolation"].getValue(), interpolation )
+				self.assertEqual( output["type"].getValue(), vectorDataType.staticTypeName() )
+				self.assertEqual( output["value"].getValue(), vectorDataType( [ value ] * mesh.variableSize( interpolation ) ) )
+
+		interpolation = IECoreScene.PrimitiveVariable.Interpolation.Constant
+		for i, ( dataType, value ) in enumerate( zip( self.dataTypes, self.values ) ) :
+			output = scriptNode["qnv"]["out"][ i ]
+			self.assertTrue( output["exists"].getValue() )
+			self.assertEqual( output["interpolation"].getValue(), interpolation )
+			self.assertEqual( output["type"].getValue(), dataType.staticTypeName() )
+			self.assertEqual( output["value"].getValue(), value )
+
+		# round trip serialise
+
+		serialised = scriptNode.serialise()
+
+		scriptNode = Gaffer.ScriptNode()
+		scriptNode.execute( serialised )
+
+		# check queries
+
+		for i, interpolation in enumerate( self.interpolations ) :
+			for j, ( vectorDataType, value ) in enumerate( zip( self.vectorDataTypes, self.values ) ) :
+				output = scriptNode["qv"]["out"][ i * len( self.names ) + j ]
+				self.assertTrue( output["exists"].getValue() )
+				self.assertEqual( output["interpolation"].getValue(), interpolation )
+				self.assertEqual( output["type"].getValue(), vectorDataType.staticTypeName() )
+				self.assertEqual( output["value"].getValue(), vectorDataType( [ value ] * mesh.variableSize( interpolation ) ) )
+
+		interpolation = IECoreScene.PrimitiveVariable.Interpolation.Constant
+		for i, ( dataType, value ) in enumerate( zip( self.dataTypes, self.values ) ) :
+			output = scriptNode["qnv"]["out"][ i ]
+			self.assertTrue( output["exists"].getValue() )
+			self.assertEqual( output["interpolation"].getValue(), interpolation )
+			self.assertEqual( output["type"].getValue(), dataType.staticTypeName() )
+			self.assertEqual( output["value"].getValue(), value )
+
```

Comment:
Would be good to add a test here for the case where the queried location doesn't exist. I'm getting a `Compute did not set plug value` exception for that here, whereas we want the compute to just output `exists = False, type = "", value = default, interpolation = invalid`.

Reply:
done